Newsletters: add 331 (2024-11-29) (#2040)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2024-11-29-newsletter.md

    [links][towns wots] to a [Twitter post][nick wots] by Jonas Nick
    about implementing Winternitz One Time Signatures (WOTS+) using
    existing opcodes and the opcodes specified in Rusty Russell's
    _great script restoration_ (GSR) [proposal][russell gsr].  Towns
    then compares implementing WOTS using symbll in bllsh.  This reduces
    the amount of data that would need to be placed onchain by at least
    83% and potentially by more than 95%.  That could allow the use of
    [quantum-safe signatures][topic quantum resistance] at a cost only
    30x greater than P2WPKH outputs.

  - *Flexible coin earmarks:* Towns [describes][towns earmarks] a
    generic construction compatible with symbll (and probably
    [Simplicity][topic simplicity]) that allows partitioning a UTXO into
    specific amounts and spending conditions.  If a spending condition
    is fulfilled, the associated amount can be spent and the remaining
    value from the UTXO is returned to a new UTXO with the remaining
    conditions.  An alternative condition may also be satisfied to allow
    spending the entire UTXO; for example, this could allow all parties

>     to agree to update some of the conditions.  This is flexible type of

Reviewer: ```suggestion
    to agree to update some of the conditions.  This is a flexible type of
```

{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

- [How does ColliderScript improve Bitcoin and what features does it enable?]({{bse}}124690)
  Victor Kolobov lists potential uses for ColliderScript (see [Newsletter
  #330][news330 cs] and [Podcast #330][pod330 cs]) including [covenants][topic
  covenants], [vaults][topic vaults], emulation of [CSFS][topic
  op_checksigfromstack], and validity rollups (see [Newsletter #222][news222
  validity rollups]) while noting the high computational costs of such transactions.

- [Why do standardness rules limit transaction weight?]({{bse}}124636)
  Murch provides arguments for and against Bitcoin Core's standardness weight
  limits and outlines how economic demand for larger transactions could erode
  the effectiveness of the [policy][policy series].

- [Is the scriptSig spending an PayToAnchor output expected to always be empty?]({{bse}}124615)
  Pieter Wuille points out that because of how [pay-to-anchor (P2A)][topic ephemeral
  anchors] outputs are [constructed][news326 p2a], they must adhere to segwit spending
  conditions, including an empty scriptSig.

- [What happens to the unused P2A outputs?]({{bse}}124617)

>   Instagibbs notes that today, unused P2A outputs with values at or above
>   Bitcoin Core's 240 satoshi [dust][topic uneconomical outputs] limit will
>   eventually be swept, removing them from the UTXO set. He goes on to reference

Reviewer 2: I don't like this wording because it suggests P2A outputs smaller than 240 sats will not be swept. How about something like:
```suggestion
  Instagibbs notes that unused P2A outputs will eventually be swept when the block
  inclusion feerate drops low enough to make a sweep worth it, removing them from
  the UTXO set. He goes on to reference
```

  the recently-merged [ephemeral dust][news330 ed] PR that allows a single
  below-dust-threshold output in a zero-fee transaction provided a [child
  transaction][topic cpfp] immediately spends it.

- [Why doesn't Bitcoin's PoW algorithm use a chain of lower-difficulty hashes?]({{bse}}124777)
  Pieter Wuille and Vojtěch Strnad describe the mining centralization pressure
  that would happen if the progress-free property of Bitcoin's mining was
  violated with such an approach.

- [Clarification on false value in Script]({{bse}}124673)
  Pieter Wuille specifies the three values that evaluate to false in Bitcoin
  Script: an empty array, an array of 0x00 bytes, and an array of 0x00 bytes with a
  0x80 at the end. He notes that all other values are evaluated as true.

- [What is this strange microtransaction in my wallet?]({{bse}}124744)
  Vojtěch Strnad explains the mechanics of an address poisoning attack and ways
  to mitigate such attacks.

> - [Are there any UTXO that can not be spent?]({{bse}}124865)

Reviewer 2: ```suggestion
- [Are there any UTXOs that can not be spent?]({{bse}}124865)
```
